File: dart/maroctax/lib/src/amo.dart
import 'fiscal_config.dart';
import 'types.dart';
import 'utils.dart';

/// Employee AMO contribution (2.26% of full gross, no cap).
double calculateAmo(double gross, [FiscalConfig? config]) {
  final cfg = config ?? getFiscalConfig();
  return roundMoney(gross * cfg.amoEmployeeRate);
}

File: typescript/maroctax/src/amo.ts
import type { FiscalConfig } from './types.js';
import { roundMoney } from './utils.js';

/**
 * Employee AMO contribution (2.26% of full gross, no cap).
 * @param gross - Monthly gross salary in MAD.
 * @param config - Fiscal configuration for the payroll year.
 */
export function calculateAmo(gross: number, config: FiscalConfig): number {
  return roundMoney(gross * config.amoEmployeeRate);
}
